src/crcspeed.c

```diff
@@ -1,11 +1,21 @@
 /*
  * Copyright (C) 2013 Mark Adler
+ * Copyright (C) 2019-2024 Josiah Carlson
  * Originally by: crc64.c Version 1.4  16 Dec 2013  Mark Adler
  * Modifications by Matt Stancliff <matt@genges.com>:
  *   - removed CRC64-specific behavior
  *   - added generation of lookup tables by parameters
  *   - removed inversion of CRC input/result
  *   - removed automatic initialization in favor of explicit initialization
+ * Modifications by Josiah Carlson <josiah.carlson@gmail.com>
+ *   - Added case/vector/AVX/+ versions of crc combine function; see crccombine.c
```

Comment:
maybe we can use avx2 instructions to accelerate, and i find a repo https://github.com/SchrodingerZhu/crc64-cxx, @Nugine could you please share your thoughts？

Reply:
The benchmark looks promising (> 20GiB/s). Modern CPUs have hardware CRC32 support and efficient instructions for CRC64. It's worth trying.

See also https://github.com/tikv/crc64fast for the instructions
> a SIMD-carryless-multiplication based implementation on modern processors:
>    using PCLMULQDQ + SSE 4.1 on x86/x86_64
>    using PMULL + NEON on AArch64 (64-bit ARM)

Improvement of scalar implementation is also valuable for general cases.

Reply:
@Nugine , @ShooterIT,
This direction is indeed very interesting! It's important to note that while these optimizations can yield significant improvements, they are specific to certain architectures. Before adding this complexity, it's essential to verify that CRC processing is still truly a bottleneck in relevant scenarios. Fortunately, in this case, a substantial effort has already gone into exploring these suggested optimization, making it a trivial decision to leverage those findings.

Reply:
yes, do agree, we should not put much energy if crc64 doesn't cost much time. I saw we made changes, we can use if there is a fast algorithms.